### PR TITLE
Split storage/mod.rs (#167)

### PR DESCRIPTION
**Pull request by hannesdejager**

### CODE REVIEW

File: src/storage/backend.rs [hunk 1]
```diff
@@ -0,0 +1,197 @@
+//! StorageBackend that uses a local filesystem, like a traditional FTP server.
```
Comment (Nonius) [inline src/storage/backend.rs:1]: I would rather name this file `storage.rs` or `storage_backend.rs` `backend.rs` is too general

Comment (hannesdejager) [reply src/storage/backend.rs:1]: will do

Comment (hannesdejager) [reply src/storage/backend.rs:1]: thinking was that if you import it you say: `use storage::backend`, hmmm

Comment (hannesdejager) [reply src/storage/backend.rs:1]: I tried to rename to storage and clippy doesn't like it:

```
    Checking libunftp v0.7.0 (/Users/hdejager/Develop/unftp/libunftp)
error: module has the same name as its containing module
 --> src/storage/mod.rs:8:1
  |
8 | pub(crate) mod storage;
  | ^^^^^^^^^^^^^^^^^^^^^^^
 
```

Comment (hannesdejager) [reply src/storage/backend.rs]: named it storage_backend

**Pull request merged**